### PR TITLE
8286304: Removal of diagnostic flag GCParallelVerificationEnabled

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  can I get your opinion about this change, if you think it's worth doing, and if so, a review for it?

If the change goes forward, I will do a release note. Since this is a diagnostic flag, there is no CSR required.

I grepped sources for that flag, it seems to be only used in these two places.

Testing: local compilation

Thanks,
  Thomas

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8286304](https://bugs.openjdk.java.net/browse/JDK-8286304): Removal of diagnostic flag GCParallelVerificationEnabled

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to [fdb5e843](https://git.openjdk.java.net/jdk/pull/8813/files/fdb5e84349267bf4de24c95816ff959691d4e3e9)
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - **Reviewer**) ⚠️ Review applies to [fdb5e843](https://git.openjdk.java.net/jdk/pull/8813/files/fdb5e84349267bf4de24c95816ff959691d4e3e9)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8813/head:pull/8813` \
`$ git checkout pull/8813`

Update a local copy of the PR: \
`$ git checkout pull/8813` \
`$ git pull https://git.openjdk.java.net/jdk pull/8813/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8813`

View PR using the GUI difftool: \
`$ git pr show -t 8813`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8813.diff">https://git.openjdk.java.net/jdk/pull/8813.diff</a>

</details>

### CODE REVIEW

**Review by kimbarrett — COMMENTED** [review]
I don't have a strong opinion either way on the name.  But we do have other
xxxEnabled options (that are product options, so harder to change).

ParallelRefProcEnabled
ParallelRefProcBalancingEnabled

If we're going to make the change proposed here, I think those should be
changed too (with the required ceremony).

**Review by albertnetymk — APPROVED** [review]

**Comment by tschatzl** [comment]
Please also review the associated release note https://bugs.openjdk.java.net/browse/JDK-8287208

File: src/hotspot/share/gc/g1/g1HeapVerifier.cpp [hunk 1]
```diff
@@ -502,8 +497,9 @@ void G1HeapVerifier::verify(VerifyOption vo) {
     ResourceMark rm;
     LogStream ls(log.error());
     _g1h->print_extended_on(&ls);
+
+    guarantee(false, "there should not have been any failures");
```
Comment (kimbarrett) [inline src/hotspot/share/gc/g1/g1HeapVerifier.cpp:501]: Seems like clearer would be `fatal(...)` rather than `guarantee(false, ...)`?

**Review by kimbarrett — APPROVED** [review]
Looks good, other than perhaps a small change to error reporting.

**Comment by tschatzl** [comment]
Thanks @kimbarrett @albertnetymk for your reviews

/integrate

**Pull request closed without merging**